### DISCUSSION TITLE
Outside layer

### DISCUSSION THREAD

**Discussion by Bebok099** [opengisch/QField#5213] [Q&A]
Hi! i'm new user of Qfieldcloud. How to add a new layer from outside to an existing project in QFieldCloud? So far, when the concept changed, I would replace the project already in the cloud with a new updated offline project (that is never been in qfieldcloud) that included the layer that needed to be added. However, this is becoming tiring with a larger number of concept changes.

**Comment by SeqLaz** [comment]
Hi @Bebok099 ,

Make sure you're saving the layer in the right spot with QFieldSync. Find the right project folder, usually at (../{user}/QField/{Project name}/). Then, copy and paste the GeoPackage into that folder and import it to the project.

https://docs.qfield.org/get-started/tutorials/get-started-qfc/

![image](https://github.com/opengisch/QField/assets/77129293/2a8cc805-34aa-4436-b6b9-13250723123c)

**Comment by Bebok099** [comment]
Hi! Thank you for your response. I did exactly as you instructed. Interestingly, the newly created GPKG layer saved in the project folder was transferred to the cloud after synchronization. However, it does not display on external devices where the project is located, despite multiple synchronizations with the cloud. I also removed the project from the device and downloaded it again, but the layer, despite being present in the cloud, still does not display. What could be the issue?

**Comment by SeqLaz** [comment]
Thank you for your response @Bebok099 

Could you provide a sample project?

**Comment by Bebok099** [comment]
Hi! 
Something strange happened. After multiple synchronizations of the project with the cloud, where the newly created layer was already present, it still did not appear on the external device. Today, I synchronized the project in the same way as when adding this layer, but to transfer changes in layers that existed from the beginning of the project, and suddenly the layer I created a week ago and couldn't see appeared. I did everything the same, and suddenly after a week, it appears in the project... Why so late, and why during this particular synchronization (conducted always the same way)? Unfortunately, our projects contain sensitive data, and I can't share them with you. If you have any ideas without needing the data, let me know. If not, thank you very much for your help!

**Reply by Ipickedausername** [reply]
Hi, 
They don't need the data, only a sample project. But, in order to better understand, they need your project. It can also be shared in private if needed.

**Reply by SeqLaz** [reply]
Hi @Bebok099,

I apologize for the late reply.

Could you try to fix the schema of the layer by using the core plugin called "Refactor field"? 

I have attached a document with a common workflow to use the plugin. Make sure that you save the newly generated GeoPackage in the QFieldCloud local project on your computer. After successfully completing the processing, copy all the styles from the original layer to the new one and paste all the styles onto it. Optionally, and as a recommendation, you could delete the old layers from the layers list and from the project folder, then synchronize the changes to the cloud. Now the layers should be shown correctly in QField. 
[Workflow fixing displaying features 001.pdf](https://github.com/user-attachments/files/15855204/Workflow.fixing.displaying.features.001.pdf)

**Reply by Bebok099** [reply]
Hi @SeqLaz!

I apologize for the late reply too!

Now everthing work good, thank you very much for your help!
